src/Features/CSharp/Portable/QuickInfo/CSharpDiagnosticAnalyzerQuickInfoProvider.cs

```diff
@@ -36,13 +31,15 @@ public CSharpDiagnosticAnalyzerQuickInfoProvider(IDiagnosticAnalyzerService diag
             _diagnosticAnalyzerService = diagnosticAnalyzerService;
         }
 
-        protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(
-            Document document,
-            SyntaxToken token,
-            CancellationToken cancellationToken)
+        protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(QuickInfoContext context)
         {
-            return GetQuickinfoForPragmaWarning(document, token) ??
-                (await GetQuickInfoForSuppressMessageAttributeAsync(document, token, cancellationToken).ConfigureAwait(false));
+            if (context is not QuickInfoContextWithDocument contextWithDocument)
+            {
+                return null;
```

Comment:
📝 This will be valuable Quick Info for LSIF, so we'll want a plan to eventually include it.

Reply:
Sure, this would likely need a compiler API where analyzers can somehow access all supported diagnostic descriptors in the compilation. I am not sure if there is an alternate solution, given LSIF generator will eventually be an analyzer/artifact producer.